core/src/Kokkos_Parallel.hpp

```diff
@@ -496,6 +496,96 @@ void parallel_scan( const std::string& str
   (void) str;
 }
 
+//VINH DANG -- Adding the following for returning final scan result
+template< class ExecutionPolicy , class FunctorType, class ReturnType >
+inline
+void parallel_scan( const ExecutionPolicy & policy
+                  , const FunctorType     & functor
+                  , const std::string& str = ""
+                  , typename Impl::enable_if< ! Impl::is_integral< ExecutionPolicy >::value >::type * = 0
+                  , ReturnType        & return_value=0
```

Comment:
`return_value` should be the third to last argument, and it should not have a default value.